test/unit/link-warnings.test.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @jest-environment jsdom
+ */
+import { act, render } from '@testing-library/react'
+import Link from 'next/link'
+
+describe('<TestComponent/>', () => {
+  let spy
+  beforeAll(() => {
+    spy = jest.spyOn(console, 'error').mockImplementation(() => {})
+  })
+
+  // Commenting out "test a" makes the problem go away.
+  // This fails with NextJS 10.x, but works with 9.5.6.
+  it('test a', () => {})
+
+  it('test c', () => {
+    act(() => {
+      render(
+        <div>
+          <p>Hello</p>
+          <Link href="nowhere">hello</Link>
+        </div>
+      )
+    })
```

Comment:
I think you need to unmount to trigger the issue. It won't be called on initial render but when the idle callback attempts to run after the component has already unmounted.